<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>
  <script>
    /*
     * then:指定用于得到成功value的成功回调和用于得到失败reason的失败的回调，并且将返回一个新的Promise实例化对象
     * 成功的状态：执行then方法的第一个回调函数
     * 失败的状态：执行then方法的第二个回调函数
     * 
     * then 方法的返回值Promise实例化对象的状态取决于回调函数中的内容
     * 如果返回为非Promise实例化对象，则得到一个是成功的Promise
     * 如果返回的是Promise实例化对象，则Promise实例化对象的状态和结果值将直接影响result
     * 如果为抛出异常，则新的Promise实例化对象（result）为失败的Promise
     */

    const p1 = new Promise((resolve, reject) => {
      resolve("ok");
      // reject('err')
    });

   const result = p1.then((value)=>{
      return new Promise((resolve,reject)=>{
        // resolve(value);
        throw new Error('异常信息')
      })
    },(reason)=>{
      return new Promise((resolve,reject)=>{
        reject('err')
      })
    })

    console.log('result',result)


  </script>
</body>
</html>